src/cloud/components/Views/index.tsx

```diff
@@ -78,6 +81,22 @@ export const ViewsManager = ({
     )
   )
 
+  useEffect(() => {
+    if (!query || typeof query.view !== 'string') {
+      return
+    }
+
+    try {
+      const viewId: number = parseInt(query.view)
+      const viewToLoad = views.find((view) => view.id === viewId)
+      if (viewToLoad != null) {
+        setSelectedViewId(viewToLoad.id)
+      }
+    } catch (_) {
+      // leave selected view as is
```

Comment:
Please check existence of the view before selecting instead of throwing an error without proper handling.